src/main/java/pages/GameDetailsPage.java

```diff
@@ -0,0 +1,152 @@
+    private static final By gamePurchasePriceBy = By.cssSelector(".game_purchase_price.price");
+    private static final By gameDiscountBy = By.className("discount_pct");
+    private static final By winPlatformIconBy = By.className("win");
+    private static final By macPlatformIconBy = By.className("mac");
+    private static final By linuxPlatformIconBy = By.className("linux");
+
+    public GameDetailsPage(BrowserService browserService) {
+        super(browserService, null);
+    }
+
+    @Override
+    protected By getIndicatorThatPageOpenedElementLocator() {
+        return pageContentBy;
+    }
+
+    public Select getAgeDropdown() {
+        return new Select(browserService.getDriver().findElement(ageDropdownBy));
+    }
+
+    public WebElement getViewPageBnt() {
+        return browserService.getDriver().findElement(viewPageBtnBy);
+    }
+
+    public By getViewPageBtnBy() {
+        return viewPageBtnBy;
+    }
+
+    public By getGameAreaPurchaseGameItemBy() {
+        return gameAreaPurchaseGameItemBy;
+    }
+
+    public WebElement getGameAreaPurchaseGameItem() {
+        return browserService.getDriver().findElements(gameAreaPurchaseGameItemBy).get(0);
+    }
+
+    public GenreCatalogGameItem getGameItemFromDetailPage() {
+
+        return new GenreCatalogGameItem(
+                getGameName(getGameAreaPurchaseGameItem()),
+                getPriceWithDiscountIfExists(getGameAreaPurchaseGameItem()),
+                getFinalPrice(getGameAreaPurchaseGameItem()),
+                getDiscountIfExists(getGameAreaPurchaseGameItem()),
+                getPlatforms(getGameAreaPurchaseGameItem())
+        );
+    }
+
+    @SneakyThrows
+    private String getGameName(WebElement gameItem){
+        try {
+            return Utils.replaceStringWithoutTradeMark(gameItem.findElement(gameNameFromPurchaseGameItemBy).getText().replaceFirst("^[a-zA-Zа-яА-я]+\\s", ""));
+        } catch (Error e) {
+            var errMes = "Couldn't get game name. \nDetailed message: \n" + e.getMessage();
+            log.error(errMes);
+            throw new Exception(errMes);
+        }
+    }
+
+    @SneakyThrows
+    private Double getPriceWithDiscountIfExists(WebElement gameItem) {
+        try {
+            if (gameItem.findElements(gameOriginalPriceBy).size() > 0) {
+                return Utils.getNumberFormString(gameItem.findElement(gameOriginalPriceBy).getText());
+            } else {
+                return null;
+            }
+        } catch (Error e) {
+            var errMes = "Couldn't find original price. \nDetailed message: \n" + e.getMessage();
+            log.error(errMes);
+            throw new Exception(errMes);
+        }
+    }
+
+    @SneakyThrows
+    private Double getFinalPrice(WebElement gameItem) {
+        try {
+            if (gameItem.findElements(gamePurchasePriceBy).size() > 0) {
+                if (gameItem.findElement(gamePurchasePriceBy).getText().contains("Free")) {
+                    return 0.0;
```

Comment:
It's a bad practice to have several return lines in 1 method. You need to have 1 return per method